juju/juju
# DDL SQL files

In order to get better IDE support for sql files, we can embed the sql files into the binary directly. This provides the best of both worlds. We can have independent sql files, that we can validate, format, and test. All whilst we read from embedded strings. There will be no disk reads when using this.

The files are read in order, so 0000-9999. This is per model, and controllers will have their independent range.

Linting is out of scope of this initial PR. We can employ a github action to perform the correct linting over the ddl files.



## Checklist




## QA steps

The existing tests should pass.

Regression testing:

```
$ juju bootstrap lxd test
$ juju add-model default
$ juju deploy ubuntu
```

## Links

**Jira card:** JUJU-
